index.js

```diff
@@ -17,46 +16,60 @@ var tree = quadtree(0, 0, gameConfig.width, gameConfig.height);
 var players = [];
 var men = [];
 var sockets = {};
-
-function addMan (manToAdd, socketId) {
-	manToAdd.id = socketId;
-	manToAdd.x = utils.getRandomNum();
-	manToAdd.y = utils.getRandomNum();
-	men.push(manToAdd);
-	tree.put(manToAdd);
-}
-
-function removeMan (manToRemove) {
-	var removeIndex = men.indexOf(manToRemove);
-	men.splice(removeIndex, 1);
-	tree.remove(manToRemove);
-}
+var removedPlayers = 0; // once it reaches 100 garbage COLLECTION!
+
+// function addMan (manToAdd, socketId) {
+// 	manToAdd.id = socketId;
+// 	manToAdd.x = utils.getRandomNum();
+// 	manToAdd.y = utils.getRandomNum();
+// 	men.push(manToAdd);
+// 	tree.put(manToAdd);
+// }
+
+// function removeMan (manToRemove) {
+// 	var removeIndex = men.indexOf(manToRemove);
+// 	men.splice(removeIndex, 1);
+// 	tree.remove(manToRemove);
+// }
 
 function addPlayer(playerData, socketId) {
-	playerData.id = socketId
+	playerData.id = socketId;
 	players.push(playerData);
 }
 
-function moveLoop () {
-	for (var i = 0; i < men.length; i++) {
-		trackMan(men[i]);
-	}
+function removePlayer (socket) {
+    removedPlayers += 1;
+    if (removedPlayers > 100) {
+    	players = utils.garbageCollection(players);
+    	removedPlayers = 0;
+    }
+   
+    delete sockets[socket.id];
+    players.forEach(function (player) {
+        if (player.id === socket.id) player = null;
+    });
 }
 
-function trackMan (man) {
-	moveMan(man);
+// function moveLoop () {
+// 	for (var i = 0; i < men.length; i++) {
+// 		trackMan(men[i]);
+// 	}
+// }
 
-	tree.clear();
-	men.forEach(tree.put);
-}
+// function trackMan (man) {
+// 	moveMan(man);
 
-function moveMan (man) {
+// 	tree.clear();
+// 	men.forEach(tree.put);
+// }
 
-}
+// function moveMan (man) {
+
+// }
 
 function sendUpdates () {
 	players.forEach(function (player) {
-		sockets[player.id].emit('gameUpdate', 'asdf');
+        if (sockets[player.id]) sockets[player.id].emit('gameUpdate', 'asdf'); 
```

Comment:
have to have this check here because sometimes after deletion, it will get sockets[player.id] is undefined.